### PR TITLE
Replace pull_request_target workflow with two workflows that upload/download an artifact.

### PR DESCRIPTION
**Pull request by dblock**
### Description

When we use `pull_request_target` we check out code that has potentially been modified by the requester, which isn't safe. This PR replaces the workflow with something inspired by https://securitylab.github.com/research/github-actions-preventing-pwn-requests/ - a gather step and a comment step that use an artifact to pass information between them.

Here's a a gather workflow that produces the artifact: https://github.com/dblock/opensearch-api-specification/actions/runs/8740901986
Here's a comment workflow that uses it: https://github.com/dblock/opensearch-api-specification/actions/runs/8740884453

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/coverage-gather.yml [hunk 1]
```diff
@@ -50,14 +48,15 @@ jobs:
...
+          {
+            "pull_request":${{ github.event.number }},
+            "current":$current,
+            "total":$total,
+            "percent":$percent
+          }
           EOL
-      - uses: peter-evans/create-or-update-comment@v4
-        if: github.event_name == 'pull_request_target'
+      - uses: actions/upload-artifact@v2
```
Comment (Xtansia) [inline .github/workflows/coverage-gather.yml:59]: `actions/upload-artifact@v2` is set to be deprecated, `v4` is the latest: https://github.blog/changelog/2024-04-16-deprecation-notice-v3-of-the-artifact-actions/

File: .github/workflows/coverage-comment.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+               owner: context.repo.owner,
+               repo: context.repo.repo,
+               artifact_id: matchArtifact.id,
+               archive_format: 'zip',
+            });
+            var fs = require('fs');
+            fs.writeFileSync('${{github.workspace}}/coverage.zip', Buffer.from(download.data));
+      - run: |
+            unzip coverage.zip
+            cat coverage.json
```
Comment (Xtansia) [inline .github/workflows/coverage-comment.yml:38]: `actions/download-artifact@v4` now supports downloading artifacts from other workflow runs: https://github.com/actions/download-artifact?tab=readme-ov-file#download-artifacts-from-other-workflow-runs-or-repositories

**Review by Xtansia — CHANGES_REQUESTED** [review]

File: .github/workflows/coverage-comment.yml (hunk 2 above)
Comment (dblock) [reply .github/workflows/coverage-comment.yml]: Updated. This is much nicer.

Successful comment run: https://github.com/dblock/opensearch-api-specification/actions/runs/8752961841
Comment on: https://github.com/dblock/opensearch-api-specification/pull/5

**Comment by dblock** [comment]
Also https://code.dblock.org/2024/04/19/commenting-on-pull-requests-in-github-actions.html.

File: .github/workflows/coverage-comment.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+          name: coverage
+          run-id: ${{ github.event.workflow_run.id }}
+
+      - name: 'Comment on PR'
+        uses: actions/github-script@v3
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const fs = require('fs');
+            var data = JSON.parse(fs.readFileSync('./coverage.json'));
```
Comment (nhtruong) [inline .github/workflows/coverage-comment.yml:29]: nit: avoid declaring a variable with `var`, as you MIGHT overwrite `var data` declared else where use `const` in this case instead.

**Review by Xtansia — APPROVED** [review]

**Pull request merged**